shared/chat/conversation/list.desktop.js

```diff
@@ -74,14 +90,24 @@ class ConversationList extends Component<void, Props, State> {
   componentWillReceiveProps (nextProps: Props) {
     // If we're not scrolling let's update our internal messages
     if (!this.state.isScrolling) {
+      this._invalidateChangedMessages()
```

Comment:
Not sure I follow this. This appears to compare `props.messages` and `this.state.messages`, but doesn't `componentWillReceiveProps` run before either has changed? It seems like this would compare the previous known states.